third_party/proto/cosmos/distribution/v1beta1/genesis.proto

```diff
@@ -0,0 +1,197 @@
+  option (gogoproto.goproto_getters) = false;
+
+  // validator_address is the address of the validator.
+  string validator_address = 1
+      [ (gogoproto.moretags) = "yaml:\"validator_address\"" ];
+
+  // accumulated is the accumulated commission of a validator.
+  ValidatorAccumulatedCommission accumulated = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"accumulated\""
+  ];
+}
+
+// ValidatorHistoricalRewardsRecord is used for import / export via genesis
+// json.
+message ValidatorHistoricalRewardsRecord {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // validator_address is the address of the validator.
+  string validator_address = 1
+      [ (gogoproto.moretags) = "yaml:\"validator_address\"" ];
+
+  // period defines the period the historical rewards apply to.
+  uint64 period = 2;
+
+  // rewards defines the historical rewards of a validator.
+  ValidatorHistoricalRewards rewards = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"rewards\""
+  ];
+}
+
+// ValidatorCurrentRewardsRecord is used for import / export via genesis json.
+message ValidatorCurrentRewardsRecord {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // validator_address is the address of the validator.
+  string validator_address = 1
+      [ (gogoproto.moretags) = "yaml:\"validator_address\"" ];
+
+  // rewards defines the current rewards of a validator.
+  ValidatorCurrentRewards rewards = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"rewards\""
+  ];
+}
+
+// DelegatorStartingInfoRecord used for import / export via genesis json.
+message DelegatorStartingInfoRecord {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // delegator_address is the address of the delegator.
+  string delegator_address = 1
+      [ (gogoproto.moretags) = "yaml:\"delegator_address\"" ];
+
+  // validator_address is the address of the validator.
+  string validator_address = 2
+      [ (gogoproto.moretags) = "yaml:\"validator_address\"" ];
+
+  // starting_info defines the starting info of a delegator.
+  DelegatorStartingInfo starting_info = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"starting_info\""
+  ];
+}
+
+// ValidatorSlashEventRecord is used for import / export via genesis json.
+message ValidatorSlashEventRecord {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // validator_address is the address of the validator.
+  string validator_address = 1
+      [ (gogoproto.moretags) = "yaml:\"validator_address\"" ];
+  // height defines the block height at which the slash event occured.
+  uint64 height = 2;
+  // period is the period of the slash event.
+  uint64 period = 3;
+  // validator_slash_event describes the slash event.
+  ValidatorSlashEvent validator_slash_event = 4
```

Comment:
for proto, does this need the same exact variable name as the go-type or the name can be arbitrary as long as the types match?

Reply:
i see that the var name is `event` in the go defined struct